### ISSUE TITLE
Suddenly getting errors being thrown because I'm using `content` as a property

### ISSUE THREAD

**Issue by mansona** [emberjs/data#7679]
### Description

I'm suddenly coming across this issue as a result of ember-data@3.28 being released, and it's clearly as a result of https://github.com/emberjs/data/pull/7545 being released finally. 

I understand the motivation for implementing something like this but surely this is a major braking change that should have had a deprecation or been implemented as a debug assert so that it wouldn't break any applications that are currently working with it. 

I know there is an argument to say that these applications aren't in fact "currently working" because of the issues related to relationships, but in the case of any application that wasn't accessing `content` through a relationship this change in a minor version of ember-data is very surprising 😳  at the very least I would expect some way to opt out of this error i.e. because you know your application won't hit the bad usecase. 

I know we're currently in the process of releasing ember@3.28 but I wonder if it would be possible to revert this change and make it a deprecation at least? I have a feeling that a lot of people will come across this issue, not least every single user of anything Empress related because every model we use has `content` in it 🙈 

### Versions

Run the following command and paste the output below: `npm list ember-source && npm list ember-cli && npm list --pattern ember-data`.

```cli
empress-blog-twitch-template@0.0.0 /Users/mansona/temp/empress-blog-twitch-template
├── ember-source@3.27.5
└─┬ empress-blog-ghost-helpers@1.0.0 -> ./../../git/opensource/empress-blog/empress-blog-ghost-helpers
  └── ember-source@3.20.3

empress-blog-twitch-template@0.0.0 /Users/mansona/temp/empress-blog-twitch-template
├── ember-cli@3.27.0
└─┬ empress-blog-ghost-helpers@1.0.0 -> ./../../git/opensource/empress-blog/empress-blog-ghost-helpers
  └── ember-cli@3.20.0

empress-blog-twitch-template@0.0.0 /Users/mansona/temp/empress-blog-twitch-template
└── ember-data@3.28.0
```

**Comment by mansona** [comment]
@igorT yes absolutely! I added one with just a failing test here https://github.com/emberjs/data/pull/7682 and a different PR that adds the test and actually reverts the commit that causes the issue: https://github.com/emberjs/data/pull/7683

**Comment by mansona** [comment]
I have closed the PR that just added the failing test now that the fix PR has been merged. I'll not yet close this issue though until the backport and release is out for 3.28 👍

**Comment by mansona** [comment]
So this was merged and released in 3.28.1 I think 🤔  but whatever version it was released in the problem is fixed now 🎉
